apps/src/codebridge/FileBrowser/prompts/openNewFolderPrompt.tsx

```diff
@@ -0,0 +1,64 @@
+import {getNextFolderId} from '@codebridge/codebridgeContext';
+import {NewFolderFunction} from '@codebridge/codebridgeContext/types';
+import {DEFAULT_FOLDER_ID} from '@codebridge/constants';
+import {ProjectType, FolderId} from '@codebridge/types';
+import {
+  sendCodebridgeAnalyticsEvent,
+  validateFolderName,
+} from '@codebridge/utils';
+
+import codebridgeI18n from '@cdo/apps/codebridge/locale';
+import {
+  DialogType,
+  DialogControlInterface,
+  extractDialogClosePromiseInput,
+} from '@cdo/apps/lab2/views/dialogs';
+import {EVENTS} from '@cdo/apps/metrics/AnalyticsConstants';
+
+type OpenNewFilePromptArgsType = {
+  parentId: FolderId;
+  appName: string;
+  dialogControl: DialogControlInterface;
+  newFolder: NewFolderFunction;
+  projectFolders: ProjectType['folders'];
+};
+
+export const openNewFolderPrompt = async ({
+  parentId = DEFAULT_FOLDER_ID,
+  appName,
+  dialogControl,
+  newFolder,
+  projectFolders,
+}: OpenNewFilePromptArgsType) => {
+  const results = await dialogControl.showDialog({
+    type: DialogType.GenericPrompt,
+    title: codebridgeI18n.newFolderPrompt(),
+    validateInput: (folderName: string) => {
+      if (!folderName.length) {
+        return;
+      }
+      if (!validateFolderName(folderName)) {
+        return codebridgeI18n.invalidNameError();
+      }
+      const existingFolder = Object.values(projectFolders).some(
+        f => f.name === folderName && f.parentId === parentId
+      );
+      if (existingFolder) {
+        return codebridgeI18n.folderExistsError();
+      }
+    },
+  });
+  if (results.type !== 'confirm') {
+    return;
+  }
+  const folderName = extractDialogClosePromiseInput(results);
+
+  const folderId = getNextFolderId(Object.values(projectFolders));
+  newFolder({parentId, folderName, folderId});
+
+  const eventName =
+    parentId === DEFAULT_FOLDER_ID
+      ? EVENTS.CODEBRIDGE_NEW_FOLDER
+      : EVENTS.CODEBRIDGE_NEW_SUBFOLDER;
+  sendCodebridgeAnalyticsEvent(eventName, appName);
+};
```

Comment:
Logically, this is same as the memoized function that used to be defined inside of `FileBrowser`, except instead of being memoized logic with a dependency array of external data going into it, it's retooled slightly to an object with all of the necessary inputs.

The intent is for all of these `open*Prompt` functions to be wrapped up with `usePartialApply` so that we ended up with functions which only accept the one or two arguments that are actually changing, and the rest just comes along for the ride for free.